go/heif/heif.go

```diff
@@ -600,32 +694,55 @@ func (img *Image) GetImage() (image.Image, error) {
 }
 
 type ImageAccess struct {
-	Plane  []byte
-	Stride int
+	Plane    []byte
+	planePtr unsafe.Pointer
+	Stride   int
+	Height   int
 
 	image *Image // need this reference to make sure the image is not GC'ed while we access it
 }
 
+func (i *ImageAccess) SetData(data []byte, stride int) {
```

Comment:
Please also require `width` and `height` to be passed and check their values (`width` and `height` must match values from `ImageAccess` object), otherwise this could lead to segfaults (if `data` is too small or `width`, `height` or `stride` are too large). Also check if the size of `data` is `stride * height`.

You can then handle the common case directly:
```go
if stride == i.Stride {
  i.Plane = data
  i.planePtr = unsafe.Pointer(&i.Plane[0])
  return
}
```

Maybe it's easier to not expose `SetData` for now if not required to be called by apps using the library.

Reply:
As discussed, I will unexport the method for now and only handle the common case (`stride == i.Stride `) without checking `width` or `height`.